Sources/DataStructuresAndAlgorithms/Algorithms/SortingAlgorithm.swift

```diff
@@ -131,25 +131,27 @@ struct SortingAlgorithm {
     }
 
     private func merge<T: Comparable>(n1: [T], n2: [T]) -> [T] {
-        var n1 = n1
-        var n2 = n2
+        let size1 = n1.count
+        let size2 = n2.count
```

Comment:
@darhonbek I don't want to use `count` for performance reasons. When you have a lot of items, it starts making a difference.